presto-hive/src/test/java/com/facebook/presto/twitter/hive/thrift/TestHiveThriftFieldIdResolver.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive.thrift;
+
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Properties;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+@Test
+public class TestHiveThriftFieldIdResolver
+{
+    private static final Map<String, Object> STRUCT_FIELD_ID_AS_MAP = ImmutableMap.of(
+            "0", (short) 1,
+            "1", (short) 2,
+            "id", (short) 4);
+
+    private static final Map<String, Object> LIST_FIELD_ID_AS_MAP = ImmutableMap.of(
+            "0", "{}",
+            "id", (short) 5);
+
+    private static final Map<String, Object> VERBOSE_PRIMARY_FIELD_ID_AS_MAP = ImmutableMap.of(
+            "id", (short) 6);
+
+    private static final Map<String, Object> THRIFT_FIELD_ID_JSON_AS_MAP = ImmutableMap.<String, Object>builder()
+            .put("0", (short) 1)
+            .put("1", (short) 3)
+            .put("2", STRUCT_FIELD_ID_AS_MAP)
+            .put("3", LIST_FIELD_ID_AS_MAP)
+            .put("4", VERBOSE_PRIMARY_FIELD_ID_AS_MAP)
+            .build();
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final ThriftFieldIdResolverFactory resolverFactory = new HiveThriftFieldIdResolverFactory();
+
+    @Test
+    public void testDefaultResolver()
+            throws Exception
+    {
+        ThriftFieldIdResolver plusOne = resolverFactory.createResolver(new Properties());
+
+        assertEquals(plusOne.getThriftId(0), 1);
+        assertEquals(plusOne.getThriftId(10), 11);
+        assertEquals(plusOne.getThriftId(5), 6);
+        assertEquals(plusOne.getNestedResolver(2), plusOne);
```

Comment:
Because the default non-informative resolver should return itself since it's nested resolver must be non-informative resolver